Source/EventFlow/Aggregates/AggregateRoot.cs

```diff
@@ -135,46 +135,49 @@ protected virtual void Emit<TEvent>(TEvent aggregateEvent, IMetadata metadata =
             return domainEvents;
         }
 
-        public void ApplyEvents(IReadOnlyCollection<IDomainEvent> domainEvents)
+        public void ApplyEvents(IReadOnlyCollection<IDomainEvent> domainEvents, int fromVersion = 0)
```

Comment:
I don't think exposing this is a good idea. If its only for snapshots, then we should move setting the version to the `SnapshotAggregateRoot` and simply remove the `Version > 0` check.

Reply:
You're right, we already have the sequential checks now for each event, so it will fail further down if you try applying event no. 1 to an aggregate that is not in Version 0, while it still supports setting the version beforehand. The additional check seems obsolete.